spec-main/api-app-spec.ts

```diff
@@ -668,16 +704,77 @@ describe('app module', () => {
+      app.setLoginItemSettings({ openAtLogin: true, path: updateExe, args: processStartArgs, enabled: true });
+      expect(app.getLoginItemSettings().openAtLogin).to.equal(false);
+      const openAtLoginTrueEnabledTrue = app.getLoginItemSettings({
+        path: updateExe,
+        args: processStartArgs
+      });
+
+      expect(openAtLoginTrueEnabledTrue.openAtLogin).to.equal(true);
+      expect(openAtLoginTrueEnabledTrue.executableWillLaunchAtLogin).to.equal(true);
+
+      app.setLoginItemSettings({ openAtLogin: true, path: updateExe, args: processStartArgs, enabled: false });
+      const openAtLoginTrueEnabledFalse = app.getLoginItemSettings({
+        path: updateExe,
+        args: processStartArgs
+      });
 
-      app.setLoginItemSettings({ openAtLogin: true, path: updateExe, args: processStartArgs });
+      expect(openAtLoginTrueEnabledFalse.openAtLogin).to.equal(true);
+      expect(openAtLoginTrueEnabledFalse.executableWillLaunchAtLogin).to.equal(false);
 
-      expect(app.getLoginItemSettings().openAtLogin).to.equal(false);
-      expect(app.getLoginItemSettings({
+      app.setLoginItemSettings({ openAtLogin: false, path: updateExe, args: processStartArgs, enabled: false });
+      const openAtLoginFalseEnabledFalse = app.getLoginItemSettings({
         path: updateExe,
         args: processStartArgs
-      }).openAtLogin).to.equal(true);
+      });
+
+      expect(openAtLoginFalseEnabledFalse.openAtLogin).to.equal(false);
+      expect(openAtLoginFalseEnabledFalse.executableWillLaunchAtLogin).to.equal(false);
+    });
+
+    ifit(process.platform === 'win32')('allows you to pass a custom name', function () {
+      app.setLoginItemSettings({ openAtLogin: true });
+      app.setLoginItemSettings({ openAtLogin: true, name: 'additionalEntry', enabled: false });
+      expect(app.getLoginItemSettings()).to.deep.equal({
+        openAtLogin: true,
+        openAsHidden: false,
+        wasOpenedAtLogin: false,
+        wasOpenedAsHidden: false,
+        restoreState: false,
+        executableWillLaunchAtLogin: true,
+        launchItems: [{
+          name: 'additionalEntry',
+          path: process.execPath,
+          args: [],
+          scope: 'user',
+          enabled: false
+        }, {
+          name: 'electron.app.Electron',
+          path: process.execPath,
+          args: [],
+          scope: 'user',
+          enabled: true
+        }]
+      });
+
+      app.setLoginItemSettings({ openAtLogin: false, name: 'additionalEntry' });
+      expect(app.getLoginItemSettings()).to.deep.equal({
+        openAtLogin: true,
+        openAsHidden: false,
+        wasOpenedAtLogin: false,
+        wasOpenedAsHidden: false,
+        restoreState: false,
+        executableWillLaunchAtLogin: true,
+        launchItems: [{
+          name: 'electron.app.Electron',
+          path: process.execPath,
+          args: [],
+          scope: 'user',
+          enabled: true
+        }]
+      });
     });
   });
 
```

Comment:
I would like to see more test cases:
1. add launch entry , but not enabled
3. add multiple launch entries and modify one by name
2. add multiple launch entries and remove one by name

Reply:
I do this in the test case 'allows you to pass a custom name'